src/java.compiler/share/classes/javax/lang/model/util/Elements.java

```diff
@@ -707,6 +707,16 @@ boolean overrides(ExecutableElement overrider, ExecutableElement overridden,
      */
     Name getName(CharSequence cs);
 
+    /**
+     * {@return {@code true} if the element has an empty name, {@code false} otherwise}
+     *
+     * @param element the element being examined
+     * @since 21
+     */
+    default boolean hasEmptyName(Element element) {
```

Comment:
Updating some suggestions from an off-list discussion, ModuleElement and PackageElement both have an isUnnamed predicate. VariableElement could get an isUnnamed predicate (as a default method returning false and an implSpec tag requiring that) or Elements.isUnnamed(VariableElement) could be added similarly.

Presumably, at least any newly-defined methods should get marked as some category of preview method.

Also, please file a issue against core-libs/javax.lang.model for javax.lang.model support for this JEP; having a separate issue makes doing the JSR 269 maintenance release work easier.

Reply:
PR updated. Thanks for the review!